app/components/AddRuleModal.tsx

```diff
@@ -1,18 +1,15 @@
 import * as React from "react";
+import * as uuid from "uuid";
 import { Modal } from "./ModalWrapper";
 
 interface AddRuleModalProps {
-  addRequest: (url: string, method: string) => void;
-  addRequestMethod: string;
-  addRequestUrl: string;
-  addRuleErrorNotify: (errorMessage: string, tabId: number) => void;
-  addRuleError: string;
+  addRequestDetails: Object;
```

Comment:
why this is mentioned as just Object. shouldn't we point this to interface of newRequest?